### PR TITLE
App not Un-Registering when trying to Force Close Nav App

### LINKED ISSUE #3174: CCB: App not Un-Registering when trying to Force Close Nav App
### Bug Report
App not Un-Registering when trying to Force Close Nav App 

### Detailed analysis:
1. Video_streaming_Lock acquired was not released when timeout and stop called from another thread happens at the same time
- When a new video packet arrives and when wakeupstreaming is called, "video_streaming_suspend" and stop_streaming locks are acquired and a new video suspend timer is created. When the new video suspend timer is created it will stop the existing timer its referencing and then starts a new one.
The timer stop call waits until the timer callback is exited. In this case, during the stop process of the timer, a timeout occured and then the timer callback was waiting on acquiring the lock "video_streaming_suspend" which was not yet released by wakeupstreaming call. This puts it in an deadlock scenario.
- Solution for this issue will be, scoping the lock to protect only the variables it was intended to protect and not the timer stop and start calls.
2. Resumption timeout callback was waiting on Policy_Manager_lock_ which was not released when acquired if resuming application is activated.
- When an resuming app was activated and at the same time the resumption callback is triggered, policy_manager_lock wont be acquired by the resumption callback as its waiting on sdlactivateapp function call to release the lock, this lock wont be released as sdlactivateapp will be waiting to acquire "queue_lock" in resumption context which is already acquired by the resumption timeout callback.
- Solution: sdlactivateapprequest will release the "policy_manager_lock_" before activating the application as it wont be needed and allows resumption callback to acquire the lock to process resumption state for other application
3. Crash observed when one thread accessing multi-frame data and other thread deleting the multi frame data due to disconnection
- When One thread is processing mutliframe, another thread can trigger unexpected disconnect and remove the multiframe, which causes the multi frame processing thread to crash. Providing synchronization will resolve this issue

##### Reproduction Steps
1. Start SDL, HMI, connect Mobile
2. Register Nav app, start streaming
3. Force Close Nav App

##### Expected Behavior
SDL should not crash

##### Observed Behavior
SDL crashed

#### OS & Version Information
* OS/Version : SYNC4
* SDL Core Version: https://github.com/smartdevicelink/sdl_core/releases/tag/5.0.1 
* Testing Against: SYNC4 Applink

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**
Fixes #3174

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
Manual testing performed on SYNC version for SDL.
Script from https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2449 

### Summary

1. video_streaming_Lock acquired was not released when timeout and stop called from another thread happens at the same time
2. resumption timeout callback was waiting on Policy_Manager_lock_ which was not released, when acquired if resuming application is activated.
3. Crash observered when one thread accessing multiframe data and other thread deleting the mutliframe data due to disconnection

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by theresalech** [comment]
@LuxoftAKutsan, can you please let us know when this is ready for re-review?

**Comment by LuxoftAKutsan** [comment]
@theresalech this work is planned. We will notify when fix with test scripts will be available.

File: src/components/application_manager/src/policies/policy_handler.cc [hunk 1]
```diff
@@ -1385,6 +1385,7 @@ void PolicyHandler::OnActivateApp(uint32_t connection_key,
   // In this case we need to activate application
   if (false == permissions.appRevoked && true == permissions.isSDLAllowed) {
     LOG4CXX_INFO(logger_, "Application will be activated");
+    sync_primitives::AutoUnlock unlock(policy_manager_lock_);
```
[the remark this answers is not in the document]
Comment (YarikMamykin) [reply src/components/application_manager/src/policies/policy_handler.cc]: @jacobkeeler build fixed with this commit https://github.com/smartdevicelink/sdl_core/pull/3374/commits/60c8e00c314e42ed0549436d530bf47e6d5d6a9f

**Comment by LitvinenkoIra** [comment]
@theresalech This PR is ready for Livio review

**Comment by LitvinenkoIra** [comment]
Full regression successfully passed on our side

**Pull request merged**